multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ArchiveEntryExtractor.java

```diff
@@ -0,0 +1,97 @@
+                                                                }
+                                                                return inflateFileContent(fileEntryProperties, fileEntryStream);
+                                                            });
+        } catch (FileStorageException e) {
+            throw new SLException(e, e.getMessage());
+        }
+    }
+
+    private FileContentToProcess toFileContentToProcess(FileEntryProperties fileEntryProperties,
+                                                        ArchiveEntryWithStreamPositions archiveEntryWithStreamPositions) {
+        return ImmutableFileContentToProcess.builder()
+                                            .guid(fileEntryProperties.getGuid())
+                                            .spaceGuid(fileEntryProperties.getSpaceGuid())
+                                            .startOffset(archiveEntryWithStreamPositions.getStartPosition())
+                                            .endOffset(archiveEntryWithStreamPositions.getEndPosition())
+                                            .build();
+    }
+
+    private byte[] inflateFileContent(FileEntryProperties fileEntryProperties, InputStream fileEntryStream) throws IOException {
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+            InflatorUtil.inflate(new EntryToInflate(fileEntryProperties.getName(),
+                                                    fileEntryProperties.getMaxFileSizeInBytes(),
+                                                    fileEntryStream),
+                                 (bytesBuffer, bytesRead) -> byteArrayOutputStream.write(bytesBuffer, 0, bytesRead));
+            return byteArrayOutputStream.toByteArray();
+        }
+    }
+
+    public void processFileEntryContent(FileEntryProperties fileEntryProperties,
+                                        ArchiveEntryWithStreamPositions archiveEntryWithStreamPositions,
+                                        ObjIntConsumer<byte[]> decompressedBytesConsumer) {
+        try {
+            fileService.consumeFileContentWithOffset(toFileContentToProcess(fileEntryProperties, archiveEntryWithStreamPositions),
+                                                     fileEntryStream -> {
+                                                         if (archiveEntryWithStreamPositions.getCompressionMethod() == ArchiveEntryWithStreamPositions.CompressionMethod.STORED) {
+                                                             byte[] buffer = new byte[BUFFER_SIZE];
+                                                             int bytesRead;
+                                                             try (
+                                                                 InputStream inputStream = new DefaultLimitedInputStream(fileEntryStream,
```

Comment:
do you close this inputstream

Reply:
yes, it is wrapped in try with resources